service/src/test/java/com/ge/predix/controller/test/PolicyManagementControllerIT.java

```diff
@@ -81,13 +83,33 @@ public void setup() throws Exception {
         this.testZone2 = new TestUtils().createTestZone("PolicyMgmtControllerIT2");
         this.zoneService.upsertZone(this.testZone);
         this.zoneService.upsertZone(this.testZone2);
-        MockSecurityContext.mockSecurityContext(this.testZone);
-        MockAcsRequestContext.mockAcsRequestContext(this.testZone);
         this.policySet = this.jsonUtils.deserializeFromFile("controller-test/complete-sample-policy-set.json",
                 PolicySet.class);
         Assert.assertNotNull(this.policySet, "complete-sample-policy-set.json file not found or invalid");
     }
 
+    @BeforeMethod
+    public void beforeMethod() {
+        MockSecurityContext.mockSecurityContext(this.testZone);
+        MockAcsRequestContext.mockAcsRequestContext(this.testZone);
+    }
+
+    @Test
+    public void policyZoneDoesNotExistException() throws Exception {
+        // NOTE: To throw a ZoneDoesNotExistException, we must ensure that the AcsRequestContext in the
+        //       SpringSecurityZoneResolver class returns a null ZoneEntity
+        MockSecurityContext.mockSecurityContext(null);
+        MockAcsRequestContext.mockAcsRequestContext(this.testZone);
+
+        String thisUri = VERSION + "/policy-set/" + this.policySet.getName();
+        // create policy-set in first zone
+        MockMvcContext putContext = this.testUtils.createWACWithCustomPUTRequestBuilder(this.wac,
+                this.testZone.getSubdomain(), thisUri);
+        putContext.getMockMvc().perform(putContext.getBuilder().contentType(MediaType.APPLICATION_JSON)
+                .content(this.objectWriter.writeValueAsString(this.policySet))).andExpect(status()
+                .isUnprocessableEntity());
```

Comment:
Same concern as above - how can we be sure that Unprocessable Entity  status was indeed caused by ZoneDoesNotExistException? We should mock exception throw.